atc/gc/pipeline_collector.go

```diff
@@ -0,0 +1,33 @@
+package gc
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager/lagerctx"
+	"github.com/concourse/concourse/atc/db"
+)
+
+type pipelineCollector struct {
+	pipelineLifecycle db.PipelineLifecycle
+}
+
+func NewPipelineCollector(pipelineLifecyle db.PipelineLifecycle) *pipelineCollector {
+	return &pipelineCollector{
+		pipelineLifecycle: pipelineLifecyle,
+	}
+}
+
+func (pc *pipelineCollector) Run(ctx context.Context) error {
+	logger := lagerctx.FromContext(ctx).Session("pipeline-collector")
+
+	logger.Debug("start")
+	defer logger.Debug("done")
+
+	_, err := pc.pipelineLifecycle.ArchiveAbandonedPipelines()
```

Comment:
Doesn't feel like it would be a useful metric for people to consume. I decided to not return an `int` anymore, just an error from `pipelineLifecycle.ArchiveAbandonedPipelines()`